packages/flutter/lib/src/material/dialog.dart

```diff
@@ -244,7 +245,7 @@ class Dialog extends StatelessWidget {
             surfaceTintColor: surfaceTintColor ?? dialogTheme.surfaceTintColor ?? defaults.surfaceTintColor,
             shape: shape ?? dialogTheme.shape ?? defaults.shape!,
             type: MaterialType.card,
-            clipBehavior: clipBehavior,
+            clipBehavior: clipBehavior ?? dialogTheme.clipBehavior ?? defaults.clipBehavior ?? Clip.none,
```

Comment:
I need the `?? Clip.none` to keep the default behavior because I only modified M2 defaults and not M2 defaults (see https://github.com/flutter/flutter/pull/147635/files#r1586126430)